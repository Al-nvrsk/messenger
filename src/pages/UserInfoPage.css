.userinfopage {
    display: flex;
    flex-direction: column;
    min-width: 600px;
    justify-content: center;
    align-items: center;
    border: 1px dashed white;
}
