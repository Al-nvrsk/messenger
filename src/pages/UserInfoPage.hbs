{{!< chatwindow}}

<div class="userinfopage">
    <img src="../assets/default user.png">
    
     {{>"UIComponent/userstat/userinfo" 
    name="Email"
    value="test@text.ru"}}

    <div>first_name</div>
    <div>second_name</div>
    <div>display_name</div>
    <div>login</div>
    <div>email</div>
    <div>phone</div>
    <img>avatar</img>

    <div>oldPassword</div>
    <div>newPassword</div>
</div>