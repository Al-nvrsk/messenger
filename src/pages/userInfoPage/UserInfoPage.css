.mainUserInfoPage {
  min-width: 600px;
  max-width: 80%;
  margin: auto;
}

.userinfopage {
  margin-top: 20px;
  display: flex;
  flex-direction: column;
  min-width: 600px;
  justify-content: center;
  align-items: center;
}

.userInfoPageBlock {
  display: flex;
  flex-direction: column;
  margin: auto;
  margin-top: 25px;
  width: 60%;
}
