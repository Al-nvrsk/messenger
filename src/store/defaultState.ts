export interface AppState {
  appIsInited: boolean
  isLoading: boolean
  isAuth: boolean
  loginFormError: string | null
  user: User | null
  chats: object[] | null
}

export const defaultState: AppState = {
  appIsInited: false,
  isLoading: false,
  isAuth: false,
  loginFormError: null,
  user: null,
  chats: null
}
