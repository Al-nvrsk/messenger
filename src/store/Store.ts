import set from 'utils/helper/set'
import EventBus from 'utils/EventBus'
import { defaultState } from './defaultState'

export enum StoreEvents {
  Updated = 'updated',
  Deleted = 'deleted'
}

class Store extends EventBus {
  private readonly state: Indexed = defaultState

  public getState (): Indexed {
    return this.state
  }

  public setState (path: string, value: unknown): void {
    set(this.state, path, value)
    this.emit(StoreEvents.Updated)
  };
}

export default new Store()
