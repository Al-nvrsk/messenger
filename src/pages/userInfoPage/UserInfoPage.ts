import Block from 'utils/Block'
import './UserInfoPage.css'
import type { AppState } from 'store/defaultState'
import { connect } from '../../utils/helper/connect'
import userInfoPageStatus from 'data/userInfoPageStatus'
import { router } from '../../index'
import store from '../../store/Store'

class UserInfoPage extends Block {
  static componentName = 'UserInfoPage'
  constructor (props: any) {
    super(props)
    this.setProps({
      store: store.getState,
      gotoBack: () => router.back(),
      changeAvatar: () => store.setState('isChangeAvatarActive', true),
      changePassword: () => store.setState('isChangePasswordActive', true),
      changeUserInfo: () => store.setState('isChangeUserInfoActive', true)

    })
  }

  render (): string {
    if (!this.props.isAuth) {
      router.go('/auth')
      throw new Error('You have to be authorized')
    } else {
      let path = ''
      if (this.props.user) {
        path = (process.env.URL ?? 'https://ya-praktikum.tech/api/v2') + `/resources${this.props.user.avatar}`
      }
      return `
      <main>
      {{{ modalChangeAvatar active = ${this.props.isChangeAvatarActive} }}}
      {{{ modalChangePassword active = ${this.props.isChangePasswordActive} }}}
      {{{modalChangeUserInfo active = ${this.props.isChangeUserInfoActive}}}}
      
        <div class = "mainUserInfoPage">
          {{{ButtonReturn onClick=gotoBack }}}
            <div class = "userinfopage">
              <img src = "${path}" alt = "avatar">
    
              ${this.props.user && (userInfoPageStatus?.map(val =>
                  `{{{ UserStat 
                      name = "${val.name}"
                      value = "${this.props.user[val.name]}"}}}`)).join(' ')}
      
            </div>
            <div class = "userInfoPageBlock">

              
                  {{{ButtonChange value = "Change avatar" onClick = changeAvatar }}}
                  {{{ButtonChange value = "Change userInfo" onClick = changeUserInfo }}}
                  {{{ButtonChange value = "Change password" onClick = changePassword }}}
                  
                  {{{ ButtonReject value = "Exit" type = "button" onClick=gotoBack }}}
                  
            </div>
        </div>
        
        
      </main>
    `
    }
  }
}

function mapStateToProps (state: AppState | Indexed): Indexed {
  return {
    user: state.user,
    isChangeAvatarActive: state.isChangeAvatarActive,
    isChangePasswordActive: state.isChangePasswordActive,
    isChangeUserInfoActive: state.isChangeUserInfoActive,
    isAuth: state.isAuth
  }
}
const withStore = connect(mapStateToProps)

export default withStore(UserInfoPage)
