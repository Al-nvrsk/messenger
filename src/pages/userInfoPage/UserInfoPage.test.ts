import '@testing-library/jest-dom'
import UserInfoPage from './UserInfoPage'
import store from 'store/Store'
import mockedRenderDOM from 'test/mockedRenderDOM'
import mockedRegisteredComp from 'test/mockedRegisteredComp'

jest.mock('utils/renderDOM')

const testUser = {
  email: 'user@test.ru',
  first_name: 'User',
  login: 'User',
  display_name: 'User',
  phone: '1234567890',
  second_name: 'UserUser'
}

describe('pages/userInfoPage', () => {
  it('should render userInfoPage', () => {
    store.setState('isAuth', true)
    store.setState('user', testUser)
    mockedRegisteredComp()
    mockedRenderDOM(new UserInfoPage(store))
    const UserInfoPageBody = document.querySelector('.userInfoPage')
    expect(UserInfoPageBody).toMatchSnapshot()
  })
})
