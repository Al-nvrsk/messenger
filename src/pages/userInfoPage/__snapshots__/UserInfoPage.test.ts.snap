// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/userInfoPage should render userInfoPage 1`] = `
<main
  class="userInfoPage"
>
  
      
  <div
    class="modalWindow"
  >
    
        
    <div
      class="modalWindow_content"
    >
      
          
      <div
        class="changeInfoInputAvatar"
      >
        
            
        <form
          id="myUserForm"
        >
          
              
          <input
            accept="image/*"
            id="avatar"
            name="avatar"
            type="file"
          />
          
            
        </form>
        
            
        <div
          class="changeInfoButton"
        >
          
              
          <button
            class="buttonAccept"
            type="button"
          >
             Save 
          </button>
          
              
          <button
            class="buttonReject"
            type="button"
          >
             Cancel 
          </button>
          
            
        </div>
        
          
      </div>
      
        
    </div>
    
      
  </div>
  
      
  <div
    class="modalWindow"
  >
    
      
    <div
      class="modalWindow_content"
    >
      

      <form
        class="authform"
      >
        
        
        <div>
                          
                
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input old Password
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="oldPassword"
                  placeholder="undefined"
                  ref=""
                  type="password"
                  value=""
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
          

                
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input new Password
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="newPassword"
                  placeholder="newPassword"
                  ref=""
                  type="password"
                  value=""
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
          
          
          <div
            class="changeInfoButton"
          >
            
        
            <button
              class="buttonAccept"
              type="submit"
            >
               Save 
            </button>
            
        
            <button
              class="buttonReject"
              type="button"
            >
               Cancel 
            </button>
            
            
          </div>
          

        </div>
        
    
      </form>
                
    </div>
    
      
  </div>
  
      
  <div
    class="modalWindow"
  >
    
    
    <div
      class="modalWindow_content"
    >
      

      <form
        class="authform"
      >
        
        
        <div>
                          
                
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input your First name
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="first_name"
                  placeholder="first_name"
                  ref=""
                  type="text"
                  value="User"
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
           
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input your Second name
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="second_name"
                  placeholder="second_name"
                  ref=""
                  type="text"
                  value="UserUser"
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
           
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input your Login
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="login"
                  placeholder="login"
                  ref=""
                  type="text"
                  value="User"
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
           
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input your email
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="email"
                  placeholder="email"
                  ref=""
                  type="text"
                  value="user@test.ru"
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
           
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input your local name for chats
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="display_name"
                  placeholder="display_name"
                  ref=""
                  type="text"
                  value="User"
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
           
          <div>
            
      
            <div
              class="controlledInput"
            >
              
      
              <div
                class="controlledInputLabel"
              >
                
        Input your phone number
      
              </div>
              
      
              <div
                class="controlledInputField"
              >
                
      
                <input
                  class="inputBase"
                  name="phone"
                  placeholder="phone"
                  ref=""
                  type="text"
                  value="1234567890"
                />
                
      
              </div>
              
        
              <hr
                class="controlledInputLine"
              />
              
      
              <div
                class="errorrs"
              >
                
      
                <div
                  class="errorBase"
                />
                
      
              </div>
              
      
            </div>
            
      
          </div>
          
          
          <div
            class="changeInfoButton"
          >
            
        
            <button
              class="buttonAccept"
              type="submit"
            >
               Save 
            </button>
            
        
            <button
              class="buttonReject"
              type="button"
            >
               Cancel 
            </button>
            
            
          </div>
          

        </div>
        
    
      </form>
                
    </div>
    
      
  </div>
  
      
        
  <div
    class="mainUserInfoPage"
  >
    
          
    <button
      class="arrowleft"
    >
        
    </button>
    
            
    <div
      class="userinfopage"
    >
      
              
      <img
        alt="avatar"
        src="https://ya-praktikum.tech/api/v2/resources"
      />
      
    
              
      <div
        class="userinfoMain"
      >
        
        
        <div
          class="userinfo"
        >
          
          
          <span
            class="userinfoNameField"
          >
             email 
          </span>
          
          
          <p
            class="userinfoValueField"
          >
             user@test.ru 
          </p>
          
        
        </div>
        

        
        <hr
          class="userinfo_line"
        />
        
      
      </div>
       
      <div
        class="userinfoMain"
      >
        
        
        <div
          class="userinfo"
        >
          
          
          <span
            class="userinfoNameField"
          >
             fist_name 
          </span>
          
          
          <p
            class="userinfoValueField"
          >
             undefined 
          </p>
          
        
        </div>
        

        
        <hr
          class="userinfo_line"
        />
        
      
      </div>
       
      <div
        class="userinfoMain"
      >
        
        
        <div
          class="userinfo"
        >
          
          
          <span
            class="userinfoNameField"
          >
             second_name 
          </span>
          
          
          <p
            class="userinfoValueField"
          >
             UserUser 
          </p>
          
        
        </div>
        

        
        <hr
          class="userinfo_line"
        />
        
      
      </div>
       
      <div
        class="userinfoMain"
      >
        
        
        <div
          class="userinfo"
        >
          
          
          <span
            class="userinfoNameField"
          >
             display_name 
          </span>
          
          
          <p
            class="userinfoValueField"
          >
             User 
          </p>
          
        
        </div>
        

        
        <hr
          class="userinfo_line"
        />
        
      
      </div>
       
      <div
        class="userinfoMain"
      >
        
        
        <div
          class="userinfo"
        >
          
          
          <span
            class="userinfoNameField"
          >
             login 
          </span>
          
          
          <p
            class="userinfoValueField"
          >
             User 
          </p>
          
        
        </div>
        

        
        <hr
          class="userinfo_line"
        />
        
      
      </div>
       
      <div
        class="userinfoMain"
      >
        
        
        <div
          class="userinfo"
        >
          
          
          <span
            class="userinfoNameField"
          >
             phone 
          </span>
          
          
          <p
            class="userinfoValueField"
          >
             1234567890 
          </p>
          
        
        </div>
        

        
        <hr
          class="userinfo_line"
        />
        
      
      </div>
      
      
            
    </div>
    
            
    <div
      class="userInfoPageBlock"
    >
      

              
                  
      <button
        class="buttonChange"
        type="button"
      >
         Change avatar 
      </button>
      
                  
      <button
        class="buttonChange"
        type="button"
      >
         Change userInfo 
      </button>
      
                  
      <button
        class="buttonChange"
        type="button"
      >
         Change password 
      </button>
      
                  
                  
      <button
        class="buttonReject"
        type="button"
      >
         Exit 
      </button>
      
                  
            
    </div>
    
        
  </div>
  
        
        
      
</main>
`;
